File: src/test/test1.cpp
//
// test1.cpp
//

#include <stdio.h>

#include "DatastoreInterface.hpp"

int main(int argc, char *argv[])
{
  // Create DataStore
  DataStore::DataGroup* const myDS1 = DataStore::CreateDataStore("myDS1");

  // get reference to a data store object.
  DataStore::DataGroup* const myDS1_2 = DataStore::GetDataStore("myDS1");

  //--------------------------------------------------

  // use top level datastore group to create group
  myDS1->CreateDataGroup("group1");

  // get reference to the group
  DataStore::DataGroup* const group1 = myDS1->GetDataGroup("group1");

  // use top level datastore group to create group, and assign to local reference
  DataStore::DataGroup* const group2 = myDS1->CreateDataGroup("group2");

  // use group to create group, and assign to local reference
  DataStore::DataGroup* const group1a = group1->CreateDataGroup("group1a");


  // iterate over groups

  for( auto obj : myDS1->GetDataObjects() ) {
    fprintf(stdout, "%s\n", obj->Name().c_str());
  }


  return 0;
}

File: src/python/test/test1.py
#
# test1.py - test datastore.
#

import numpy as np
import datastore as ds

myDS1 = ds.create_data_store("myDS1")
print("myDS1 =", myDS1)

#a = np.array([1,2,3,4])
a = 10

obj1 = ds.add_object(myDS1, "var1", a)
print("obj1", obj1)

obj1a = myDS1.var1
print("myDS1.var1", obj1a)

if obj1 is obj1a:
    print("Same object")
else:
    print("Different objects", id(obj1), id(obj1a))
